src/main/java/com/whichclasses/frontend/SearchApiServlet.java

```diff
@@ -29,8 +35,31 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
     response.setStatus(HttpServletResponse.SC_OK);
     response.setContentType("application/json");
-    // TODO: actually implement this
-    response.getWriter().write("{\"results\":[{\"identifier\":{\"department\":\"ECE\",\"course_number\":\"351A\"},\"title\":\"All The Circuits\"}]}");
+    String quality = request.getParameter("quality");
+    String entityType = request.getParameter("entityType");
+    String department = request.getParameter("department");
+    
+    Question question = Question.OVERALL_COURSE_RATING;
+    switch(quality) {
+    case "best":
```

Comment:
Style nit: indent case statements (https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s4.8.4-switch)

In general, you'll find this project following the Google Java style guide.